server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -510,7 +510,27 @@ public static void validateV2TemplateRequest(Metadata metadata, String name, Com
             .filter(componentTemplate -> componentTemplates.containsKey(componentTemplate) == false)
             .toList();
 
-        if (missingComponentTemplates.size() > 0) {
+        // TODO: Should the code be split up in two parts for better error messages? If ignore_missing_component_templates is not set
+        // it throws a different error?
+        final List<String> ignoreMissingComponentTemplates = template.getIgnoreMissingComponentTemplates();
+
+        // Check
+        if (ignoreMissingComponentTemplates != null && ignoreMissingComponentTemplates.size() > 0) {
+            for (String missingComponentTemplate : missingComponentTemplates) {
+                // TODO if ignoreMissingComponentTemplates not set, jump directly out
+                if (ignoreMissingComponentTemplates.contains(missingComponentTemplate) == false) {
+                    throw new InvalidIndexTemplateException(
+                        name,
+                        // TODO: should we make the code more complex and only error out in the end to have the full list reported?
+                        "index template ["
+                            + name
+                            + "] specifies a missing component templates ["
+                            + missingComponentTemplate
+                            + "] that does not exist and is not part of 'ignore_missing_component_templates'"
```

Comment:
Proposal:

* If ignore_missing_component_template is not empty or not used at all, we don't leak it 
* If it is used with a non empty list but template is not found, we leave a not in the logs.

I assume if someone uses `ignore_missing_component_template` but a template does not match, it is likely a bug which will help us to quicker figure out potential issues.

Reply:
> If it is used with a non empty list but template is not found, we leave a not in the logs.

I'm not sure what you mean by this, if it has a non-empty list and the template is not found, we should throw an exception like this currently does (rather than logging anything).

Reply:
I meant exceptions, but often these exception eventually end up in the logs which I think will also be here the case.

I updated the code quite a bit. Current commit still contians the commented out code but will remove it shortyl.